pkg/asset/manifests/infrastructure.go

```diff
@@ -153,6 +153,9 @@ func (i *Infrastructure) Generate(dependencies asset.Parents) error {
 			APIServerInternalIPs: installConfig.Config.Platform.BareMetal.APIVIPs,
 			IngressIPs:           installConfig.Config.Platform.BareMetal.IngressVIPs,
 		}
+		if installConfig.Config.BareMetal.LoadBalancer != nil {
+			config.Status.PlatformStatus.BareMetal.LoadBalancer = installConfig.Config.BareMetal.LoadBalancer
```

Comment:
Shouldn't the installer always fill the load balancer config into the PlatformStatus? We should set a default for installConfig.Config.BareMetal.LoadBalancer so that we it's always set and remove the above test.

Reply:
Since the feature is in TechPreview and it's not in the stable CRD, we have a validation later that checks for the install-config field to be set only if we are deploying with TechPreview featureSet. Therefore, I think we need this check for now and we should be able to remove it once it's promoted to stable.